frontends/pytorch/python/torch_mlir_utils/codegen/torch_signature_ods_gen.py

```diff
@@ -0,0 +1,472 @@
+        # Arg types/flags.
+        arg_types = self._format_cpp_str_initlist(
+            [t[0] for t in arg_type_flags])
+        self.print(f"m.addArgTypes({arg_types});")
+        arg_flags = self._format_cpp_kvc_initlist(
+            [t[1] for t in arg_type_flags])
+        self.print(f"m.addArgConversions({arg_flags});")
+        # Returns types/flags.
+        ret_types = self._format_cpp_str_initlist(
+            [t[0] for t in return_type_flags])
+        self.print(f"m.addReturnTypes({ret_types});")
+        ret_flags = self._format_cpp_kvc_initlist(
+            [t[1] for t in return_type_flags])
+        self.print(f"m.addReturnConversions({ret_flags});")
+        self.print("return m;")
+      self.print("})();")
+      self.print("return metadata;")
+    self.print("}")
+
+  def _format_cpp_str_initlist(self, strings):
+    quoted = [self.quote(s) for s in strings]
+    joined = ", ".join(quoted)
+    return "{" + joined + "}"
+
+  def _format_cpp_kvc_initlist(self, const_name_lists):
+
+    def or_flags(flag_names):
+      if not flag_names:
+        return "KVC::kNone"
+      return "|".join([f"KVC::{n}" for n in flag_names])
+
+    or_d = [or_flags(l) for l in const_name_lists]
+    joined = ", ".join(or_d)
+    return "{" + joined + "}"
+
+
+def snakecase_to_camelcase(ident: str):
+  return "".join(x.capitalize() or "_" for x in re.split(r"[\._]", ident))
+
+
+def _load_ops_as_dict():
+  # Returns a list of dicts, each with a name that is a tuple of the form:
+  #   (kernel_name, variant)
+  # The variant may be the empty string.
+  # Concat into "{kernel_name}.{variant}", omitting the dot if variant is empty.
+  reg_ops_list = get_registered_ops()
+  reg_ops = dict()
+  for reg_op in reg_ops_list:
+    kernel_name, variant = reg_op["name"]
+    if variant:
+      reg_ops[f"{kernel_name}.{variant}"] = reg_op
+    else:
+      reg_ops[kernel_name] = reg_op
+  return reg_ops
+
+
+def _get_main_module_name():
+  return sys.modules["__main__"].__loader__.name
+
+
+ODS_BANNER = "\n".join([
+    "//===-------------------------------------------------------*- tablegen -*-===//",
+    "//",
+    "// This file is licensed under the Apache License v2.0 with LLVM Exceptions.",
+    "// See https://llvm.org/LICENSE.txt for license information.",
+    "// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception",
+    "//",
+    "// Operation summaries and descriptions were systematically derived from public",
+    "// API docstrings and are licensed accordingly:",
+    "//   https://github.com/pytorch/pytorch/blob/master/LICENSE",
+    "//===----------------------------------------------------------------------===//",
+    "// This file is automatically generated.  Please do not edit.",
+    "// Generated via:",
+    f"//   python -m {_get_main_module_name()}",
+    "//===----------------------------------------------------------------------===//",
+    "",
+    "",
+])
+
+ODS_IMPL_BANNER = "\n".join([
```

Comment:
same comment here about "ods impl" confusing naming.

Reply:
Switched to CCImplEmitter and associated names.